src/azure-cli-testsdk/azure/cli/testsdk/checkers.py

```diff
@@ -23,7 +24,12 @@ def __call__(self, execution_result):
         except jmespath.exceptions.JMESPathTypeError:
             raise JMESPathCheckAssertionError(self._query, self._expected_result, actual_result,
                                               execution_result.output)
-        if actual_result != self._expected_result and str(actual_result) != str(self._expected_result):
+        if self._case_sensitive:
+            equals = actual_result == self._expected_result or str(actual_result) == str(self._expected_result)
+        else:
+            equals = actual_result == self._expected_result \
```

Comment:
I `lower()` is also needed here.

Reply:
The type of actual_result and self._expected_result may not be string. If actual_result == self._expected_result, the check should always pass no matter case_sensitive is true or false.